scripts/publish-docs.sh

```diff
@@ -79,10 +81,7 @@ fi
 
 pushd gh-pages
 # Do not override older javadoc with anotra's placeholder:
-for file in $(git diff --name-only | grep -v "servicetalk/SNAPSHOT/javadoc/index.html" | \
-  grep -v "servicetalk/$version/javadoc/index.html"); do
-    git checkout -- $file
-done
+git diff --name-only | grep 'javadoc/index.html' | xargs git checkout --
```

Comment:
Thanks for simplifying with `xargs` 👍 
However, you have to keep excluding `servicetalk/SNAPSHOT/javadoc/index.html` and `servicetalk/$version/javadoc/index.html, otherwise they won't be updated and the old version from the git history will be recovered.

Reply:
This code seems to have updated the currently published docs AFAICT:

Docs:
https://apple.github.io/servicetalk/servicetalk/SNAPSHOT/javadoc/index.html
https://apple.github.io/servicetalk/servicetalk/0.21/javadoc/index.html

Commit:
https://github.com/apple/servicetalk/commit/b9ab3f02c5ac66a2e0256a4cbdcb157ce742c182

Reply:
FYI this was not a simplification, this was actually the bug ;)

Reply:
I don't think so, your `rm -rf` fixes the issue described in the `modifications` section as
> Fix an issue with javadocs for SNAPSHOT not cleaning up older versions of removed or renamed classes

And this is important fix, thank for catching it.

The `servicetalk/SNAPSHOT/javadoc/index.html` and `servicetalk/$version/javadoc/index.html` will be updated by that `rm` and the following `cp`.

If you won't exclude them from this `git checkout --`, git will reset all changes for `SNAPSHOT/javadoc/index.html` (it won't be updated, git will restore the old version).

In your first commit (before these changes) I see that `SNAPSHOT/javadoc/index.html` was updated correctly: https://github.com/apple/servicetalk/commit/b9ab3f02c5ac66a2e0256a4cbdcb157ce742c182#diff-09ce3dd0f73c5e798b81c37a99da5b94
In the following commit (https://github.com/apple/servicetalk/commit/08c2e3b57fc84f83e5a40752bf73674091fd203c) I see that the timestamp was not updated in `SNAPSHOT/javadoc/index.html`, because you did `git checkout --` for it. However, other files in SNAPSHOT javadoc has updated timestamp: https://github.com/apple/servicetalk/commit/08c2e3b57fc84f83e5a40752bf73674091fd203c#diff-03104bc52fa54b8040ffd814d55d6402
This is not correct. We have to exclude `servicetalk/SNAPSHOT/javadoc/index.html` and `servicetalk/$version/javadoc/index.html` from `git checkout --`.

Reply:
I understand the intent of this section of the script, however it was resetting too much and that was the bug. 

The timestamp doesn't update because we don't run `clean`, hence you'll see the same timestamps in both commits. 

You are correct however that we still need to filter `SNAPSHOT` and `$version`, it worked out well in my earlier testing due to the previous build output not revealing any changes for those files.